src/Css/Style.php

```diff
@@ -1446,6 +1493,16 @@ public function set_prop(string $prop, $val, bool $important = false, bool $clea
                 $val = self::$_defaults[$prop];
             }
 
+            // Do not resolve CSS custom property values when setting them,
+            // because the required custom property might not have been set yet,
+            // or need to be derived from a parent. Let the compute() method
+            // resolve this when the value is needed. Just set the property now.
+            if (is_string($val) && strpos($val, 'var(') !== false) {
+                $this->_props[$prop] = $val;
+                self::$_dependent_props[$prop] = true;
```

Comment:
Flagging the value as being dependent doesn't really seem to do anything by itself and removing from the code doesn't seem to have any impact. Further comments made in the previous discussion.

Reply:
Reverted to setting the value only and then return. Resolving this thread because there are more extended threads regarding dependencies.